packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -57,8 +42,16 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
   private readonly handlers = {
     fetch: (request: RequestArguments) => fetchRPCRequest(request, this.chain),
 
-    sign: (request: RequestArguments) =>
-      this.signRequestHandler.handleRequest(request, this.accounts),
+    sign: (request: RequestArguments) => {
+      if (request.method === 'eth_requestAccounts') {
+        if (this.connected) return this.accounts;
+      } else if (!this.connected) {
+        throw standardErrors.provider.unauthorized(
+          "Must call 'eth_requestAccounts' before other methods"
+        );
+      }
+      return this.signRequestHandler.handleRequest(request);
+    },
```

Comment:
it kinda took me quite some time to understand this block tho, i'm leaning towards to not have `else if`
```suggestion
    sign: (request: RequestArguments) => {
      if (request.method === 'eth_requestAccounts') {
        if (this.connected) return this.accounts;
      };
      if (!this.connected && request.method !== 'eth_requestAccounts') {
        throw standardErrors.provider.unauthorized(
          "Must call 'eth_requestAccounts' before other methods"
        );
      };
      return this.signRequestHandler.handleRequest(request);
    },
```

Reply:
linter can't think like human